src/kernel/include/thd.h

```diff
@@ -18,8 +18,22 @@
 #include "tcap.h"
 #include "list.h"
 
-struct invstk_entry {
+/* 
+ * 1. This data structure is designed to replace the comp_info used in invstk_entry.
+ * The reason for using comp_invstk_info instead of comp_info is we want to set a flag
+ * into the captbl pointer of comp_info to achieve better alignment.
+ * 2. We plan to set a bit into the captbl pointer because captbls are perfectly aligned 
+ * which means the last few bits of captbl is always "0". By this design, we can achieve
+ * both better cache line alignment and structure alignment.
```

Comment:
I'm not sure I understand this comment.. 

If we have it "perfectly aligned", why would the last few bits in the captbl be "0"?? "perfectly aligned" I read it as, the size of the captbl is a multiple of the cache line size or is a single cache line.. so there is no padding in which case, why would you have "always 0" in the last few bits??